### DISCUSSION TITLE
Complete_loop feature

### DISCUSSION THREAD

**Discussion by torannn** [jeertmans/manim-slides#252] [Ideas]
I've seen a similar project named Manim Editor. I think the feature complete_loop is quite useful. Currently, if we use the loop and press the next button, the animation will end immediately. It creates an unsmooth transition between animations. I think we can create the complete_loop feature similar to Manim Editor. When the user press the next button, the complete_loop force the animation to be finished, then continue running the next animation.
You can check out the example created by manim editor here 
https://manimcommunity.github.io/manim_editor/Tutorial/index.html

**Comment by jeertmans** [comment]
That a good idea! I will create a feature request from this :-)

**Reply by torannn** [reply]
Glad to hear that :D. I want to contribute to manim-slides but due to my lack of experience with programming, this is as much as I can do.

**Reply by jeertmans** [reply]
No problem, giving idea is great!

however, if you want to improve your programming skills, do not hesitate to give it a try! That’s how we learn things ;)

**Reply by torannn** [reply]
I've just finished a bachelor's degree in math education. Now, it's time for other things. Currently, I am learning programming from scratch. XD

**Reply by jeertmans** [reply]
No problem haha, take your time :D

**Reply by jeertmans** [reply]
For triage purposes, this has been implemented as part of v5: https://github.com/jeertmans/manim-slides/releases/tag/v5.0.0.
